arch/renesas/src/rx65n/rx65n_rspi_sw.c

```diff
@@ -0,0 +1,2154 @@
+ *            of words.  The wordsize is determined by the number of
+ *            bits-per-word selected for the SPI interface.  If nbits <= 8,
+ *            the data is packed into uint8_t's; if nbits >8, the data is
+ *            packed into uint16_t's
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+#ifndef CONFIG_SPI_EXCHANGE
+static void rspi_recvblock(FAR struct spi_dev_s *dev, FAR void *buffer,
+                          size_t nwords)
+{
+  struct rx65n_rspidev_s *priv = (struct rx65n_rspidev_s *)dev;
+  rspi_transfer(priv, NULL, buffer, nwords);
+}
+#endif
+
+/****************************************************************************
+ * Function Name: rspi_interrupt_init
+ * Description  : Configure ICU for RSPI interrupt
+ * Arguments    : none
+ * Return Value : none
+ ****************************************************************************/
+
+void rspi_interrupt_init(FAR struct rx65n_rspidev_s *priv, uint8_t bus)
+{
+  /* Enable error interrupt source bit */
+
+  IEN(ICU, GROUPAL0) = 0;  /* Disable Group AL0 interrupts */
+  IR(ICU, GROUPAL0)  = 0;  /* Clear interrupt flag */
+  IPR(ICU, GROUPAL0) = RX65N_RSPI_INTRRUPT_PRIO;
+  IEN(ICU, GROUPAL0) = 1;  /* Enable Group AL0 interrupt */
+
+#ifdef CONFIG_RX65N_RSPI0
+  if (bus == RX65N_RSPI_CHANNEL0)
+    {
+      /* Configure Transmit empty buffer interrupt source bit */
+
+      IEN(RSPI0, SPTI0) = 0;  /* Disable SPTI0 interrupts */
+      IR(RSPI0, SPTI0)  = 0;  /* Clear interrupt flag */
+      IPR(RSPI0, SPTI0) = RX65N_RSPI_INTRRUPT_PRIO;
+      IEN(RSPI0, SPTI0) = 1;  /* Enable SPTI0 interrupt */
+      ICU.IER[04].BIT.IEN7 = 1;
+
+      /* Configure Receive buffer full interrupt source bit */
+
+      IEN(RSPI0, SPRI0) = 0; /* Disable SPRI0 interrupts */
+      IR(RSPI0, SPRI0) = 0;  /* Clear interrupt flag */
+      IPR(RSPI0, SPRI0) = RX65N_RSPI_INTRRUPT_PRIO;
+      IEN(RSPI0, SPRI0) = 1; /* Enable SPRI0 interrupt */
+      ICU.IER[04].BIT.IEN6 = 1;
+
+      /* Enable error interrupt source bit */
+
+      ICU.GENAL0.BIT.EN17 = 1;
+
+      /* Enable Idle interrupt source bit */
+
+      ICU.GENAL0.BIT.EN16 = 1;
+    }
+#endif
+
+#ifdef CONFIG_RX65N_RSPI1
+  if (bus == RX65N_RSPI_CHANNEL1)
+    {
+      /* Configure Transmit empty buffer interrupt */
+
+      IEN(RSPI1, SPTI1) = 0;  /* Disable SPTI1 interrupts */
+      IR(RSPI1, SPTI1)  = 0;  /* Clear interrupt flag */
+      IPR(RSPI1, SPTI1) = RX65N_RSPI_INTRRUPT_PRIO;
+      IEN(RSPI1, SPTI1) = 1;  /* Enable SPTI1 interrupt */
+      ICU.IER[05].BIT.IEN1 = 1;
+
+      /* Configure Receive buffer full interrupt */
+
+      IEN(RSPI1, SPRI1) = 0;  /* Disable SPRI1 interrupts */
+      IR(RSPI1,  SPRI1) = 0;  /* Clear interrupt flag */
+      IPR(RSPI1, SPRI1) = RX65N_RSPI_INTRRUPT_PRIO;
+      IEN(RSPI1, SPRI1) = 1;  /* Enable SPRI1 interrupt */
+      ICU.IER[05].BIT.IEN0 = 1;
```

Comment:
ditto

Reply:
Thank you for the comments.

The comments have been addressed in 285d16d.